src/advisors/simd.ts

```diff
@@ -48,6 +52,103 @@ class SimdAdvisor implements IAdvisor {
     };
   }
 
+  private _getSuggestionExample(): SuggestionExample {
+    const before = `#ifdef _WIN32
+#include <intrin.h>
+#else
+#include <x86intrin.h>
+#endif`;
+    const after = `#ifdef _WIN32
+#include <intrin.h>
+#else
+/* carefully comment or remove if it is not used */
+// #include <x86intrin.h>
+#endif`;
+    return new SuggestionExample(before, after);
+  }
+
+  private async _generateSimdStaticScanAdvise(
+    proj: IProject,
+    req: PlainAdviseRequest
+  ): Promise<IAdviseResult> {
+    const buildConfig = proj.config.getBuildConfigForTarget(proj.config.target);
+
+    const supportedInstructionSetsObj = {
+      sse: {
+        header: "#include <xmmintrin.h>",
+        flag: "-msse",
+      },
+      sse2: { header: "#include <emmintrin.h>", flag: "-msse2" },
+      sse3: { header: "#include <pmmintrin.h>", flag: "-msse3" },
+      ssse3: { header: "#include <tmmintrin.h>", flag: "-mssse3" },
+      sse4_1: { header: "#include <smmintrin.h>", flag: "-msse4.1" },
+      sse4_2: { header: "#include <nmmintrin.h>", flag: "-msse4.2" },
+      avx: { header: "#include <immintrin.h>", flag: "-mavx" },
+    };
+
+    let setKey: keyof typeof supportedInstructionSetsObj;
+    const actions: IAction[] = [];
+
+    for (setKey in supportedInstructionSetsObj) {
+      const matched = await H.findPatternInFiles(
+        supportedInstructionSetsObj[setKey].header,
+        proj.root,
+        [C.buildDir, C.dependencyDir]
+      );
+      if (
+        matched.length &&
+        !buildConfig.getEnv("cflags").includes(supportedInstructionSetsObj[setKey].flag)
+      ) {
+        // this instruction set is used in codebase, add
+        // corresponding flag into the compiler flags.
+        const addCflags: IArg[] = [
+          {
+            option: supportedInstructionSetsObj[setKey].flag,
+            value: null,
+            type: "replace",
+          },
+        ];
+
+        const action = new ConfigEnvChangeAction(
+          proj,
+          `Add corresponding compiler flag since \`${setKey}\` instruction set is used in the codebase`,
+          {
+            cflags: addCflags,
+          }
+        );
+
+        actions.push(action);
+      }
+    }
+
```

Comment:
Maybe we should also check the `simd` option value to determine if we need to enable the `simd` option at the same time when actions array is not empty.